cluster_mapping_rules.go

```diff
@@ -68,55 +63,50 @@ func NewClusterConfigChange(c Cluster) ClusterConfigChange {
 	}
 }
 
-// A Mapping is a set of rules mapping shards onto a storage cluster.
-type Mapping interface {
+// A ClusterMappingRuleProvider provides cluster mapping rules
+type ClusterMappingRuleProvider interface {
 	// RulesForShard returns the mapping rules that currently apply to the given shard, plus
 	// a channel that can be used to receive new mapping rules
-	RulesForShard(n int) ([]MappingRule, <-chan []MappingRule, error)
-
-	// Clusters returns all of the currently defined clusters, plus a channel
-	// that can be used to receive cluster changes (new clusters, changes to
-	// cluster configurations)
-	Clusters() ([]Cluster, <-chan ClusterConfigChange, error)
+	RulesForShard(n int) ([]ClusterMappingRule, <-chan []ClusterMappingRule, error)
```

Comment:
Whose responsibility is it to close the channel return out of interest?  If it is the provider how do we signal to the provider to close the channel?  The pattern, which is also not great but seems more common, for receiving topology updates in M3DB we have is to pass in the channel as a subscriber and then the consumer of updates can close the channel (and they created it too so its symmetric).
Also: should we make shard `uint32` to match other APIs for shards?